deepseg.cc

```diff
@@ -278,6 +286,8 @@ int main(int argc, char* argv[]) {
 
 	cap.set(CV_CAP_PROP_FRAME_WIDTH,  width);
 	cap.set(CV_CAP_PROP_FRAME_HEIGHT, height);
+	if (fmt.size() > 3)
+		cap.set(CV_CAP_PROP_FOURCC, *(uint32_t*)(fmt.data()));
```

Comment:
Allow for both 4 bytes (raw) and 8 hex digits variants to be provided. Check should be on these exact lengths.

Reply:
Oh right, it was "YUYV" not "YUY". I recalled incorrectly.

> Allow for both 4 bytes (raw) and 8 hex digits variants to be provided.

Do you happen to have any documentation on the allowed format?

Reply:
I've been working from the OpenCV docs: https://docs.opencv.org/3.2.0/dd/de7/group__videoio.html - your version may vary as we've discovered. I also find I have to frequently dive into the source: https://github.com/opencv/opencv/tree/3.2.0, and in particular the `videoio` module: https://github.com/opencv/opencv/tree/3.2.0/modules/videoio/src and `cap_v4l.cpp` especially to find out what assumptions it makes, or if it's a leaky abstraction (it often is).

Reply:
Regarding the two formats: Providing the code as 4 raw bytes might interfere with character encodings or `NUL` bytes in the format code. Thus a plain `uint32_t` might work for most common codes, but it is still advisable to just handle some format that doesn't suffer from this issue and allows for all possible codes to be represented.

So sticking to your example: `MJPG` and `4D4A5047` and `4d4a5047` are identical (note the big endian for the hex).

Reply:
It would seem there are also 3 byte raw variants and thus can be less-than-8-hex-digit-variants. I hope I've addressed all possibilities in my latest change. Please re-open if not.

Reply:
> Some minor notes:
> 
>     1. check for `in.size() == 8`. This avoids ambiguity

My reasoning was: people might skip leading zeros. But I don't have any strong preferences.

>     2. the hex-format part is interpreted in the wrong byte-order. Note my previous example on how to provide MJPG in hex. The current source will fail with that example.

Maybe stupid question: why do you think `MJPG` should be represented as `0x4D4A5047`?

From what I can tell, people[1,2] expect  `0x47504A4D` (which is what you'd get when translating by hand `ord('M')  | ord('J') << 8 | ord('P') << 16 | ord('G') << 24`)

I also saw this calculation in ffmpeg sources (which I linked in the code). 

* [1] https://gix.github.io/media-types/
* [2] https://stackoverflow.com/questions/54740012/error-while-saving-video-as-mp4-using-videowriteropencv-in-ubuntu